### DISCUSSION TITLE
Setting up the development environment in windows

### DISCUSSION THREAD

**Discussion by dave-cran** [gaphor/gaphor#545] [Q&A] [answered]
Hello,

I have been running through the process of setting up the development environment in windows as described in the docs, but cant seem to setup the venv in msys2.

> $ source venv
Are we on Macos with Homebrew installed? No
Are we on MSys (Windows)? Yes
Checking: python3 --version
Checking: pkg-config --print-errors --exists gobject-introspection-1.0 >= 1.46.0
Installing Gaphor in the virtualenv...
Requirement already up-to-date: poetry==1.0.10 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (1.0.10)
Requirement already satisfied, skipping upgrade: keyring<21.0.0,>=20.0.1; python_version >= "3.5" and python_version < "4.0" in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (20.0.1)
Requirement already satisfied, skipping upgrade: tomlkit<0.6.0,>=0.5.11 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.5.11)
Requirement already satisfied, skipping upgrade: clikit<0.5.0,>=0.4.2 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.4.3)
Requirement already satisfied, skipping upgrade: pexpect<5.0.0,>=4.7.0 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (4.8.0)
Requirement already satisfied, skipping upgrade: pkginfo<2.0,>=1.4 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (1.6.1)
Requirement already satisfied, skipping upgrade: html5lib<2.0,>=1.0 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (1.1)
Requirement already satisfied, skipping upgrade: requests<3.0,>=2.18 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (2.25.0)
Requirement already satisfied, skipping upgrade: pyparsing<3.0,>=2.2 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (2.4.7)
Requirement already satisfied, skipping upgrade: cachecontrol[filecache]<0.13.0,>=0.12.4 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.12.6)
Requirement already satisfied, skipping upgrade: requests-toolbelt<0.9.0,>=0.8.0 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.8.0)
Requirement already satisfied, skipping upgrade: pyrsistent<0.15.0,>=0.14.2 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.14.11)
Requirement already satisfied, skipping upgrade: cachy<0.4.0,>=0.3.0 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.3.0)
Requirement already satisfied, skipping upgrade: shellingham<2.0,>=1.1 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (1.3.2)
Requirement already satisfied, skipping upgrade: jsonschema<4.0,>=3.1 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (3.2.0)
Requirement already satisfied, skipping upgrade: cleo<0.8.0,>=0.7.6 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from poetry==1.0.10) (0.7.6)
Requirement already satisfied, skipping upgrade: pywin32-ctypes!=0.1.0,!=0.1.1; sys_platform == "win32" in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from keyring<21.0.0,>=20.0.1; python_version >= "3.5" and python_version < "4.0"->poetry==1.0.10) (0.2.0)
Requirement already satisfied, skipping upgrade: pastel<0.3.0,>=0.2.0 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from clikit<0.5.0,>=0.4.2->poetry==1.0.10) (0.2.1)
Requirement already satisfied, skipping upgrade: pylev<2.0,>=1.3 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from clikit<0.5.0,>=0.4.2->poetry==1.0.10) (1.3.0)
Requirement already satisfied, skipping upgrade: ptyprocess>=0.5 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from pexpect<5.0.0,>=4.7.0->poetry==1.0.10) (0.6.0)
Requirement already satisfied, skipping upgrade: six>=1.9 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from html5lib<2.0,>=1.0->poetry==1.0.10) (1.15.0)
Requirement already satisfied, skipping upgrade: webencodings in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from html5lib<2.0,>=1.0->poetry==1.0.10) (0.5.1)
Requirement already satisfied, skipping upgrade: chardet<4,>=3.0.2 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from requests<3.0,>=2.18->poetry==1.0.10) (3.0.4)
Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from requests<3.0,>=2.18->poetry==1.0.10) (2020.12.5)
Requirement already satisfied, skipping upgrade: urllib3<1.27,>=1.21.1 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from requests<3.0,>=2.18->poetry==1.0.10) (1.26.2)
Requirement already satisfied, skipping upgrade: idna<3,>=2.5 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from requests<3.0,>=2.18->poetry==1.0.10) (2.10)
Requirement already satisfied, skipping upgrade: msgpack>=0.5.2 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from cachecontrol[filecache]<0.13.0,>=0.12.4->poetry==1.0.10) (1.0.0)
Requirement already satisfied, skipping upgrade: lockfile>=0.9; extra == "filecache" in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from cachecontrol[filecache]<0.13.0,>=0.12.4->poetry==1.0.10) (0.12.2)
Requirement already satisfied, skipping upgrade: attrs>=17.4.0 in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from jsonschema<4.0,>=3.1->poetry==1.0.10) (20.3.0)
Requirement already satisfied, skipping upgrade: setuptools in c:/msys64/home/dave/gaphor/.venv/lib/python3.8/site-packages (from jsonschema<4.0,>=3.1->poetry==1.0.10) (49.2.1)
WARNING: You are using pip version 20.2.1; however, version 20.3.1 is available.
You should consider upgrading via the 'C:/msys64/home/dave/gaphor/.venv/bin/python3.exe -m pip install --upgrade pip' command.
The virtual environment found in C:/msys64/home/Dave/gaphor/.venv seems to be broken.
Recreating virtualenv gaphor-T5SV8fW6-py3.8 in C:/Users/Dave/AppData/Local/pypoetry/Cache/virtualenvs/gaphor-T5SV8fW6-py3.8
Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = (not set)
  program name = 'C:/Users/Dave/AppData/Local/pypoetry/Cache/virtualenvs/gaphor-T5SV8fW6-py3.8/bin/python3.exe'
  isolated = 1
  environment = 0
  user site = 0
  import site = 1
  sys._base_executable = 'C:\\Users\\Dave\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\gaphor-T5SV8fW6-py3.8\\bin\\python3.exe'
  sys.base_prefix = 'D:\\a\\_temp\\msys\\msys64\\mingw64'
  sys.base_exec_prefix = 'D:\\a\\_temp\\msys\\msys64\\mingw64'
  sys.executable = 'C:\\Users\\Dave\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\gaphor-T5SV8fW6-py3.8\\bin\\python3.exe'
  sys.prefix = 'D:\\a\\_temp\\msys\\msys64\\mingw64'
  sys.exec_prefix = 'D:\\a\\_temp\\msys\\msys64\\mingw64'
  sys.path = [
    'D:\\a\\_temp\\msys\\msys64\\mingw64\\lib\\python38.zip',
    'D:\\a\\_temp\\msys\\msys64\\mingw64\\lib\\python3.8',
    'D:\\a\\_temp\\msys\\msys64\\mingw64\\lib\\python3.8',
    'D:\\a\\_temp\\msys\\msys64\\mingw64\\lib\\lib-dynload',
  ]
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'
Current thread 0x000016b0 (most recent call first):
<no Python frame>.
[CalledProcessError]
Command '['C:/Users/Dave/AppData/Local/pypoetry/Cache/virtualenvs/gaphor-T5SV8fW6-py3.8/bin/python3.exe', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
PyGobject3 (gobject-introspection) can not be loaded.
The Command used to test this:
    >>> import gi
Please read the installation instructions on https://gaphor.readthedocs.io.

Hope you can help with this.

**Comment by danyeaw** [comment]
Hi @dave-cran, I noticed that you are running your shell from `C:/msys64`, but then the sys paths are all `D:\a\_temp\msys2\msys64`. Do you know why that is?

**Reply by dave-cran** [reply]
Not sure why. I suspect this switch happens after:

> The virtual environment found in C:/msys64/home/Dave/gaphor/.venv seems to be broken.
Recreating virtualenv gaphor-T5SV8fW6-py3.8 in C:/Users/Dave/AppData/Local/pypoetry/Cache/virtualenvs/gaphor-T5SV8fW6-py3.8

I have been looking at the venv shell script, but cant really figure this out

**Comment by dave-cran** [comment, marked as answer]
This issue can be solved as described here:
https://github.com/python-poetry/poetry/issues/2867

**Comment by danyeaw** [comment]
@dave-cran I think it is also a workaround to have poetry not manage your virtualenv by setting:
```
poetry config virtualenvs.create false
```
